src/Symfony/Component/Console/Application.php

```diff
@@ -76,17 +81,20 @@ class Application
     /**
      * Constructor.
      *
-     * @param string $name    The name of the application
-     * @param string $version The version of the application
+     * @param string                   $name            The name of the application
+     * @param string                   $version         The version of the application
+     * @param CommandResolverInterface $commandResolver
      */
-    public function __construct($name = 'UNKNOWN', $version = 'UNKNOWN')
+    public function __construct($name = 'UNKNOWN', $version = 'UNKNOWN', CommandResolverInterface $commandResolver = null)
     {
         $this->name = $name;
         $this->version = $version;
         $this->defaultCommand = 'list';
         $this->helperSet = $this->getDefaultHelperSet();
         $this->definition = $this->getDefaultInputDefinition();
 
+        $this->commandResolver = $commandResolver !== null ? $commandResolver : new CommandResolver();
```

Comment:
I think we use `$this->commandResolver = $commandResolver ?: new CommandResolver()`